src/Common/src/Interop/Windows/mincore/Interop.SHGetKnownFolderPath.cs

```diff
@@ -0,0 +1,320 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class mincore
+    {
+        internal const int COR_E_PLATFORMNOTSUPPORTED = unchecked((int)0x80131539);
+
+        // https://msdn.microsoft.com/en-us/library/windows/desktop/bb762188.aspx
+        [DllImport(Libraries.ShellFolders, CharSet = CharSet.Unicode, SetLastError = false, BestFitMapping = false, ExactSpelling = true)]
+        internal static extern int SHGetKnownFolderPath(
+            [MarshalAs(UnmanagedType.LPStruct)] Guid rfid,
```

Comment:
[MarshalAs(LpStruct)] Guid does additional copy internally, while ref GUID passes directly by-ref, so ref Guid is slightly better. Also, [in] ref Guid is the better option than ref Guid since ref Guid is [in, out] ref Guid and implies marshalling in both directions, which isn't the case here. 
In this case, either way it makes no practical difference.